CMakeLists.txt

```diff
@@ -540,3 +540,11 @@ string(REGEX REPLACE "/" "\\\\\\\\" NSIS_IMAGE_PATH ${NSIS_IMAGE_PATH})
 set(CPACK_NSIS_INSTALLER_MUI_ICON_CODE "!define MUI_WELCOMEFINISHPAGE_BITMAP \\\"${NSIS_IMAGE_PATH}sidebar.bmp\\\"\n!define MUI_HEADERIMAGE_BITMAP \\\"${NSIS_IMAGE_PATH}header.bmp\\\"\n!define MUI_ICON \\\"${NSIS_IMAGE_PATH}sfml.ico\\\"")
 
 include(CPack)
+
+if(NOT CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+    return()
+endif()
+
+add_custom_target(format
+    COMMAND clang-format -i `git ls-files *.hpp *.inl *.cpp`
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
```

Comment:
Something we still need to solve is crafting a shell command that will find all files ending in .hpp, .inl, and .cpp but not if they're in the extlibs directory. As it stands now if any .hpp, .inl, or .cpp files are added to extlibs, they will get formatted as well which is no good.

Reply:
There's another problem where `git ls-files` finds files ending in .hpp or .cpp that also contain spaces in the filenames which them breaks when clang-format tries to interpret that filename as two files which don't actually exist. I like using `git ls-files` since it has a better chance of working in a Windows shell but we may have to use Unix shell commands that reduce portability but are more correct.

Reply:
Turns out git has its own wildcard processor, and we can exclude directories within ls-files using `':!:extlibs/*`.

To solve the issue of spaces, pipe to xargs instead of using substitution. Unfortunately that does limit us to *nix, but I don't think that should be a problem, since we only need to run clang-format on one platform.